src/sentry/explore/translation/discover_translation.py

```diff
@@ -0,0 +1,226 @@
+    orderby_converted_list = _format_orderby_for_translation(orderby, columns)
+
+    # TODO(nikki): return dropped fields from translate_mep_to_eap
+    translated_query_parts = translate_mep_to_eap(
+        QueryParts(
+            selected_columns=columns,
+            query=conditions,
+            equations=equations,
+            orderby=orderby_converted_list,
+        )
+    )
+
+    translated_aggregate_columns = (
+        [field for field in translated_query_parts["selected_columns"] if is_aggregate(field)]
+        if translated_query_parts["selected_columns"] is not None
+        else []
+    )
+
+    translated_non_aggregate_columns = (
+        [field for field in translated_query_parts["selected_columns"] if not is_aggregate(field)]
+        if translated_query_parts["selected_columns"] is not None
+        else []
+    )
+
+    translated_equations = (
+        [field for field in translated_query_parts["equations"]]
+        if translated_query_parts["equations"] is not None
+        else []
+    )
+
+    # if we have any aggregates or equation we should be in aggregate mode
+    if len(translated_aggregate_columns) > 0 or len(translated_equations) > 0:
+        mode = "aggregate"
+    else:
+        mode = "samples"
+
+    display = query.get("display", "default")
+    interval = None
+
+    chart_type = CHART_TYPES[display]
+    # only intervals that matter are the daily ones, rest can be defaulted to explore default
+    if display in ["daily", "dailytop5"]:
+        interval = "1d"
+
+    y_axes = translated_aggregate_columns + translated_equations
+    # aggregate fields parameter contains groupBys and yAxes
+    aggregate_fields = [
+        {"groupBy": translated_column} for translated_column in translated_non_aggregate_columns
+    ] + [{"yAxes": [y_axis], "chartType": chart_type} for y_axis in y_axes]
+
+    # we want to make sure the id field is always included in samples mode
+    # because without it the 'id' field is not sortable on the samples table
+    fields_with_id = (
+        (["id"] + translated_non_aggregate_columns)
+        if "id" not in translated_non_aggregate_columns
+        else translated_non_aggregate_columns
+    )
+
+    if translated_query_parts["orderby"] is None or len(translated_query_parts["orderby"]) == 0:
+        translated_orderby = None
+        aggregate_orderby = None
+    else:
+        translated_orderby = translated_query_parts["orderby"][0]
+        stripped_translated_orderby, _ = strip_negative_from_orderby(translated_orderby)
+        aggregate_orderby = (
+            translated_orderby
+            if is_aggregate(stripped_translated_orderby) or is_equation(stripped_translated_orderby)
+            else None
+        )
+
+    query_list = [
+        {
+            "query": translated_query_parts["query"],
+            "fields": fields_with_id,
+            "orderby": (translated_orderby if aggregate_orderby is None else None),
```

Comment:
I'm not quite sure I follow what's happening here with translated_orderby and aggregate_orderby